pkg/addons/manifest.go

```diff
@@ -340,3 +360,107 @@ func vSphereCSIWebhookConfigTemplateFunc() (string, error) {
+				VolumeAttributes: map[string]string{
+					"secretProviderClass": secretProviderClassName,
+				},
+			},
+		},
+	}
+
+	volumeMount := corev1.VolumeMount{
+		Name:      "secrets-store",
+		MountPath: "/mnt/secrets-store",
+		ReadOnly:  true,
+	}
+
+	for i := range docs {
+		ubject := metav1unstructured.Unstructured{}
+		_, _, err := metav1unstructured.UnstructuredJSONScheme.Decode(docs[i].Raw, nil, &ubject)
+		if err != nil {
+			return err
+		}
+
+		switch ubject.GroupVersionKind().GroupKind() {
+		case appsv1.SchemeGroupVersion.WithKind("Deployment").GroupKind():
+			var obj appsv1.Deployment
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.Template.Spec, volume, volumeMount)
+			})
+		case appsv1.SchemeGroupVersion.WithKind("StatefulSet").GroupKind():
+			var obj appsv1.StatefulSet
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.Template.Spec, volume, volumeMount)
+			})
+		case appsv1.SchemeGroupVersion.WithKind("DaemonSet").GroupKind():
+			var obj appsv1.DaemonSet
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.Template.Spec, volume, volumeMount)
+			})
+		case appsv1.SchemeGroupVersion.WithKind("ReplicaSet").GroupKind():
+			var obj appsv1.ReplicaSet
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.Template.Spec, volume, volumeMount)
+			})
+		case corev1.SchemeGroupVersion.WithKind("Pod").GroupKind():
+			var obj corev1.Pod
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec = addVolumeToPodSpec(obj.Spec, volume, volumeMount)
+			})
+		case batchv1.SchemeGroupVersion.WithKind("Job").GroupKind():
+			var obj batchv1.Job
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.Template.Spec, volume, volumeMount)
+			})
+		case batchv1.SchemeGroupVersion.WithKind("CronJob").GroupKind():
+			var obj batchv1.CronJob
+			err = repackObject(&obj, &docs[i], func() {
+				obj.Spec.JobTemplate.Spec.Template.Spec = addVolumeToPodSpec(obj.Spec.JobTemplate.Spec.Template.Spec, volume, volumeMount)
+			})
+		}
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func repackObject(kubeobject runtime.Object, obj *runtime.RawExtension, mutator func()) error {
+	if err := yaml.Unmarshal(obj.Raw, kubeobject); err != nil {
+		return err
+	}
+
+	mutator()
+
+	buf, err := yaml.Marshal(kubeobject)
+	if err != nil {
+		return err
+	}
+
+	js, err := yaml.YAMLToJSON(buf)
+	if err != nil {
+		return err
+	}
+
+	obj.Raw = js
+
+	return nil
+}
+
+func addVolumeToPodSpec(podSpec corev1.PodSpec, volume corev1.Volume, volumeMount corev1.VolumeMount) corev1.PodSpec {
+	podSpec.Volumes = append(podSpec.Volumes, volume)
+	for i := range podSpec.Containers {
+		podSpec.Containers[i].VolumeMounts = append(podSpec.Containers[i].VolumeMounts, volumeMount)
```

Comment:
This has an unwanted security implication. Not all containers need access to the credentials. For example, CSI driver has 5-7 containers, but only one container needs credentials. That's mostly like okay for the beginning, but we should keep in mind that this might now work for production/audited setups.